Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java

```diff
@@ -119,12 +119,12 @@ public boolean equals(Object obj) {
   public static AnalysisError fromJson(JsonObject jsonObject) {
     String severity = jsonObject.get("severity").getAsString();
     String type = jsonObject.get("type").getAsString();
-    String code = jsonObject.get("code") == null ? null : jsonObject.get("code").getAsString();
     Location location = Location.fromJson(jsonObject.get("location").getAsJsonObject());
     String message = jsonObject.get("message").getAsString();
     String correction = jsonObject.get("correction") == null ? null : jsonObject.get("correction").getAsString();
+    String code = jsonObject.get("code").getAsString();
```

Comment:
Making it optional works for me since it addresses the immediate but it's a bit of a hack.  In the long run we probably need to have a better means to evolve server protocols.  Adding new data really shouldn't blow up clients IMO.

Anyway, hopping of the soapbox, if making this optional works, I say go for it! 👍